test/integration/create_classes_test.go

```diff
@@ -0,0 +1,243 @@
+				if err != nil {
+					return err
+				}
+				s := &databasev1beta1.CloudSQLInstanceClass{}
+				if err := yaml.Unmarshal(dat, s); err != nil {
+					return err
+				}
+
+				defer func() {
+					if err := c.Delete(context.Background(), s); err != nil {
+						t.Error(err)
+					}
+				}()
+
+				return c.Create(context.Background(), s)
+			},
+		},
+		"CreateMySQLCloudSQLClass": {
+			reason: "A GCP MySQL CloudSQLClass should be created without error.",
+			test: func(c client.Client) error {
+				dat, err := ioutil.ReadFile("../../examples/database/mysqlinstance/resource-class.yaml")
+				if err != nil {
+					return err
+				}
+				s := &databasev1beta1.CloudSQLInstanceClass{}
+				if err := yaml.Unmarshal(dat, s); err != nil {
+					return err
+				}
+
+				defer func() {
+					if err := c.Delete(context.Background(), s); err != nil {
+						t.Error(err)
+					}
+				}()
+
+				return c.Create(context.Background(), s)
+			},
+		},
+		"CreateV1Beta1GKEClusterClass": {
+			reason: "A v1beta1 GCP GKEClusterClass should be created without error.",
+			test: func(c client.Client) error {
+				dat, err := ioutil.ReadFile("../../examples/container/kubernetescluster/resource-class.yaml")
+				if err != nil {
+					return err
+				}
+				s := &containerv1beta1.GKEClusterClass{}
+				if err := yaml.Unmarshal(dat, s); err != nil {
+					return err
+				}
+
+				defer func() {
+					if err := c.Delete(context.Background(), s); err != nil {
+						t.Error(err)
+					}
+				}()
+
+				return c.Create(context.Background(), s)
+			},
+		},
+		"CreateV1Alpha1GKEClusterClass": {
+			reason: "A v1alpha3 GCP GKEClusterClass should be created without error.",
+			test: func(c client.Client) error {
+				dat, err := ioutil.ReadFile("../../examples/compute/kubernetescluster/resource-class.yaml")
+				if err != nil {
+					return err
+				}
+				s := &computev1alpha3.GKEClusterClass{}
+				if err := yaml.Unmarshal(dat, s); err != nil {
+					return err
+				}
+
+				defer func() {
+					if err := c.Delete(context.Background(), s); err != nil {
+						t.Error(err)
+					}
+				}()
+
+				return c.Create(context.Background(), s)
+			},
+		},
+		"CreateCloudMemorystoreInstanceClass": {
+			reason: "A GCP CloudMemorystoreInstanceClass should be created without error.",
+			test: func(c client.Client) error {
+				dat, err := ioutil.ReadFile("../../examples/cache/rediscluster/resource-class.yaml")
+				if err != nil {
+					return err
+				}
+				s := &cachev1beta1.CloudMemorystoreInstanceClass{}
+				if err := yaml.Unmarshal(dat, s); err != nil {
+					return err
+				}
+
+				defer func() {
+					if err := c.Delete(context.Background(), s); err != nil {
+						t.Error(err)
+					}
+				}()
+
+				return c.Create(context.Background(), s)
+			},
+		},
+	}
+
+	cfg, err := clientcmd.BuildConfigFromFlags("", "../../kubeconfig.yaml")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", "../../sa.json"); err != nil {
```

Comment:
Note: we look for these specific files that Jenkins drops into the project directory. Falling back on local `kubeconfig` was discussed but given that these test could eventually spin up new infrastructure, I actually like that a specific file path is hard coded.